chainercv/datasets/siamese_dataset.py

```diff
@@ -0,0 +1,119 @@
+import collections
+import numpy as np
+
+import chainer
+
+
+def _construct_label_to_key(labels):
+    d = collections.defaultdict(list)
+    for i, label in enumerate(labels):
+        d[label].append(i)
+    return d
+
+
+class SiameseDataset(chainer.dataset.DatasetMixin):
+
+    """A dataset that returns samples fetched from two datasets.
+
+    The dataset returns samples from the two base datasets.
+    If :obj:`pos_ratio` is not :obj:`None`,
+    :class:`SiameseDataset` can be configured to return positive
+    pairs at the ratio of :obj:`pos_ratio` and negative pairs at the ratio
+    of :obj:`1 - pos_ratio`.
+    In this mode, the base datasets are assumed to be label datasets that
+    return an image and a label as a sample.
+
+    Args:
+        dataset_0: The first base dataset.
+        dataset_1: The second base dataset.
+        pos_ratio (None or float): If this is not :obj:`None`,
+            this dataset tries to construct positive pairs at the
+            given rate. If :obj:`None`,
+            this dataset randomly samples examples from the base
+            datasets. The default value is :obj:`None`.
+        length (None): The length of this dataset. If :obj:`None`,
```

Comment:
`length (int)`
I think we can omit default `None` form the type description. i.e. `pos_ratio (None or float)` -> `pos_ratio (float)`